### ISSUE TITLE
Add mixed mode for react/jsx-curly-newline

### ISSUE THREAD

**Issue by Dron007** [jsx-eslint/eslint-plugin-react#3539]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

My rules are based on the Standard rules and I want to have these styles as **correct** for single line components:

```js
<div>
  {someCondition &&
    <Comp1>test</Comp1>
  }
</div>


<div>
  {someCondition && <Comp1>test</Comp1>}
</div>
```
and this as **incorrect** style
```js
<div>
  {someCondition &&
    <Comp1>test</Comp1>}
</div>
```
For multiline components I want thsse styles as **corrrect**:
```js
<div>
  {someCondition &&
    <Comp1>
      <Comp1>test</Comp1>
      <Comp1>test</Comp1>
    </Comp1>
  }
</div>

<div>
  {
    someCondition &&
      <Comp1>
        <Comp1>test</Comp1>
        <Comp1>test</Comp1>
      </Comp1>
  }
</div>
```
Closing curly bracket should be placed below the opening one. Otherwise it is hard to find it in big components.

Standard has this rule:
```js
    "react/jsx-curly-newline": [
      "error",
      {
        "multiline": "consistent",
        "singleline": "consistent"
      }
    ],
```
which moves closing bracket at the end of the last string if there was no line after the opening bracket. I didn't find any appropriate option in `react/jsx-curly-newline` rule. There are "consistent" | "forbid" | "require" but I don't need a newline after the opening bracket and a newline before the closing bracket should exist only when both condition and component are on the same line.

I also noticed that I can wrap the component with parenthesis like this 

```js
<div>
  {someCondition && (
    <Comp1>
      <Comp1>test</Comp1>
      <Comp1>test</Comp1>
    </Comp1>
  )}
</div>
```
which is also quite appropriate for multiline but it doesn't force this style and autofix doen't work.

I see there is also a `react/jsx-wrap-multilines` rule and if I set its 'logical' option to 'parens-new-line'. it does what I need but only for multiline components. How can I forbid incorrect styles for single-line components too and make autofix work? 

### Expected Behavior

see above

### eslint-plugin-react version

v7.32.2

### eslint version

v8.34.0

### node version

v14.17.1

**Comment by ljharb** [comment]
I certainly like the styles you want to be correct :-)

I'm open to suggestions for what rule changes would be needed to cover it.

**Comment by Dron007** [comment]
I don't know all details of implementation and rule dependencies. Currently I have Standard rules and override them with:
```js
    'react/jsx-curly-newline': 0,
    'react/jsx-wrap-multilines': [
      'error',
      {
        'declaration': 'parens-new-line',
        'assignment': 'parens-new-line',
        'return': 'parens-new-line',
        'arrow': 'ignore',
        'condition': 'parens-new-line', // 'ignore' in Standard
        'logical': 'parens-new-line', // 'ignore' in Standard
        'prop': 'ignore',
      },
```
Any options that forbid incorrect styles for single-line components and make autofix work will fit me.
